### PR TITLE
Add loaders and remove actions.

### PR DESCRIPTION
**Pull request by jamen**
I've implemented this new feature in light of @seanc's ideas.  I'm calling them "loaders".  There is a page in the docs I added called ["Loaders"](https://github.com/Devcord/cordlr/blob/e3b09ed252d4f9cd7ba2bee582eff99cb5ef0c6a/docs/loader.md), you can read all about them there.

In terms of changes, loaders will be a seamless transition because everything is backwards compatible, yet adds some cool functionality.

In addition, I've cleaned up the code in general, and removed `setup` and `install` (as we are transitioning to a more simple and powerful state, rather than robust and incomplete/useless)

Additions list:
- The `"loader"`property in your config, and module type for it.
- CLI aliases `x`, `p`, `t`, and `l` to `prefix`, `plugins`, `token`, and `loader` respectively, and `plugin`
- Supplying a folder scope as first argument, i.e. `cordlr foo` will load `./foo/package.json`. (This replaces actions)
- Support for `"email"` and `"password"` alternative to `"token"` (Not recommended, but convenient)

Removal list:
- `cordlr install`: it is simplest to rely on `npm install` (the user already has it and is likely familiar).
- `cordlr setup`: We only have `"loader"` natively now, and you can use other package.json generators.
- `cordlr help` and `cordlr version`: Not standard, use `--help` and `--version` flags instead.

### CODE REVIEW

**Comment by jamen** [comment]
new usage looks like this:

``` sh
cordlr 
  -t "MTA1MTAzMpQ8LTU3MTf4Njcx.aaiYWg.e0nhn6UvUlwCNgeE8CMAc2k2Bsg"
  -p cordlr-color
  -p cordlr-svg
  -p chordlr
```

``` sh
cordlr preset-config-1
  --email='Blah blah'
  --password='asd.asdasd'
  -x '/' # Prefix
```

**Comment by jamen** [comment]
This also helps with trivial / opinionated / experimental things.  If you have an idea on how plugins should work, pop in a new loader and test your ideas with code (maybe even publish it to npm):

``` js
"loader": "./my-experimental-loader"
```

This can help with compatibility in other bot frameworks too. (if there are any.  I'm not aware of any)

**Comment by mees-** [comment]
if we won't have a setup command we should maybe think about creating a yeoman generator to maintain easy setup for less experienced users

File: bin/cordlr [hunk 1]
```diff
@@ -3,34 +3,34 @@ const minimist = require('minimist')
...
-main()
 
 function main () {
-  // --help, -h, help
-  if (flags.help || flags.h || action === 'help') return help()
-
-  // --version, -v, version
-  if (flags.version || flags.v || action === 'version') return version()
+  // --help, -h
+  if (flags.help) return help()
```
Comment (mees-) [inline bin/cordlr:26]: I agree with replacing it with the flags, i think it conforms to the standard better

File: bin/loader.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+        // Run it, if it is valid
+        if (commands.has(command)) {
+          commands.get(command)(message, args, flags)
+        }
+      }
+    })
+
+    // Emit "loaded" for setup complete
+    bot.emit('loaded', !!plugins.length || new Error('No plugins'))
```
Comment (mees-) [inline bin/loader.js]: I don't think one event(name) should be used for succes and error, i think emitting an error event if no plugins are specified would be better

File: bin/methods/start.js [hunk 3]
```diff
@@ -1,12 +1,26 @@
...
+  let config = {}
+  const configPath = path.resolve(process.cwd(), folder, 'package.json')
+  try { config = require(configPath) } catch (e) {}
+
+  // Add command-line options to config
+  Object.assign(config, flags)
+  delete config._
+
+  // Attach method to write config to file
+  config.writeToFile = (cb) => writeFile(configPath, JSON.stringify(config, null, 2), cb)
```
Comment (mees-) [inline bin/methods/start.js]: maybe use a promise here, see #74

**Review by mees- — CHANGES_REQUESTED** [review]

**Comment by jamen** [comment]
I also have an idea for alternative usage in loading plugins:

``` js
cordlr cordlr-color cordlr-svg chordlr -t ...
```

Instead of needing the `-p` flag.  I also don't think scoped folder is very useful compared to this.

**Comment by jamen** [comment]
Branch has been updated.

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "cordlr",
-  "description": "Cordlr Bot for Discord",
+  "description": "A discord bot using plugins created from the community.",
```
Comment (mees-) [inline package.json]: Shouldn't this be `created by the community`?

Comment (jamen) [reply package.json]: Updated.

**Review by mees- — APPROVED** [review]
Looks good, i like the new loader idea!

**Pull request merged**